### PR TITLE
Fix array to string conversion in Navigation block server side rendering

### PR DESCRIPTION
**Pull request by draganescu**
## Description
Currently there is a bug in master where the server side rendering of the Navigation block does an array to string conversion and that generates a Notice.

## How has this been tested?
Tested locally,

## Types of changes
I `implode` the array in a string with each element glued by an empty space. The notice is gone.

### CODE REVIEW

**Comment by draganescu** [comment]
One problem here is that now I get:

```
TypeError: "_ref8 is undefined"
    ColorDetector use-colors.js:205
    useMemo react-dom.min.b694e242.js:205
    useMemo react.min.0212dc62.js:32
    __experimentalUseColors use-colors.js:202
    Navigation edit.js:56
```

File: packages/block-library/src/navigation/index.php [hunk 1]
```diff
@@ -76,8 +76,8 @@ function render_block_navigation( $attributes, $content, $block ) {
  */
 function build_navigation_html( $block, $colors ) {
 	$html = '';
-
-	$class_attribute = sprintf( ' class="%s"', esc_attr( $colors['css_classes'] ? 'wp-block-navigation-item__link ' . $colors['css_classes'] : 'wp-block-navigation-item__link' ) );
+	$css_classes = implode( ' ', $colors['css_classes'] );
+	$class_attribute = sprintf( ' class="%s"', esc_attr( ! empty ( $css_classes ) ? 'wp-block-navigation-item__link ' . $css_classes : 'wp-block-navigation-item__link' ) );
```
Comment (jorgefilipecosta) [inline packages/block-library/src/navigation/index.php]: Space is not required here "empty (" and is a lint error.

File: packages/block-library/src/navigation/index.php [hunk 2]
```diff
@@ -76,8 +76,8 @@ function render_block_navigation( $attributes, $content, $block ) {
  */
 function build_navigation_html( $block, $colors ) {
 	$html = '';
-
-	$class_attribute = sprintf( ' class="%s"', esc_attr( $colors['css_classes'] ? 'wp-block-navigation-item__link ' . $colors['css_classes'] : 'wp-block-navigation-item__link' ) );
+	$css_classes = implode( ' ', $colors['css_classes'] );
```
Comment (jorgefilipecosta) [inline packages/block-library/src/navigation/index.php]: With these changes, we are already adding space between classes so we need to remove the space from ' has-%s-color' in:
$colors['css_classes'][] = sprintf( ' has-%s-color', $attributes['textColor'] );

**Review by jorgefilipecosta — APPROVED** [review]
The PR looks good to me and can be merged after the comments are addressed 👍

**Pull request merged**